### PR TITLE
Add gerbil evaluation service

### PR DESCRIPTION
**Pull request by DnlKI**
Perform and save GERBIL evaluations of our QA systems automatically at startup.
Visualize GERBIL results of our QA system on a website.

### CODE REVIEW

File: KBQA/kbqa/webservice/gerbil/app/templates/base.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<html>
+  <head>
+    <title>Evaluation - Knowledge Base Question Answering</title>
+    <link
+      rel="stylesheet"
+      href="{{ url_for('static', filename='css/main.css') }}"
+    />
+    <script src="{{ url_for('static', filename='js/main.js') }}"></script>
```
Comment (smeyer198) [inline KBQA/kbqa/webservice/gerbil/app/templates/base.html:8]:
```suggestion
```
If main.js is deleted, this line should also be removed.

File: KBQA/kbqa/webservice/gerbil/app/templates/index.html [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      rel="stylesheet"
+      href="{{ url_for('static', filename='css/main.css') }}"
+    />
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
+      crossorigin="anonymous"
+    />
+    <script src="{{ url_for('static', filename='js/main.js') }}"></script>
```
Comment (smeyer198) [inline KBQA/kbqa/webservice/gerbil/app/templates/index.html:14]:
```suggestion
```
If main.js is deleted, this line should also be removed.

File: KBQA/kbqa/webservice/gerbil/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+flask
```
Comment (smeyer198) [inline KBQA/kbqa/webservice/gerbil/requirements.txt:1]:
```suggestion
flask==2.0.2
```

File: KBQA/kbqa/webservice/gerbil/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+requests
```
Comment (smeyer198) [inline KBQA/kbqa/webservice/gerbil/requirements.txt:2]:
```suggestion
requests==2.26.0
```

File: KBQA/kbqa/webservice/gerbil/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+uwsgi
```
Comment (smeyer198) [inline KBQA/kbqa/webservice/gerbil/requirements.txt:3]:
```suggestion
uwsgi==2.0.20

File: KBQA/kbqa/docker-compose.yml [hunk 6]
```diff
@@ -33,6 +33,15 @@ services:
...
       - 24802
 
+  gerbil:
+    build: ./webservice/gerbil
+    container_name: gerbil
+    restart: always
+    expose:
+      - 24804
+    volumes:
+      - ./gerbil/evaluation:/evaluation
```
Comment (smeyer198) [inline KBQA/kbqa/docker-compose.yml:43]: I don't know, which path is preferred, but we can change the path to "${RESOURCE_PATH}/gerbil/evaluation:/evaluation" just like the other volumes to have it consistent. You can decide on that.

**Review by smeyer198 — COMMENTED** [review]
I think the main.js can be deleted, because it is actually never used and already used by the website_server :D
I also added hooks for this subproject to the pre-commit s.t. Pylint and mypy will also check the files. You probably want to update some parts according to the errors

File: KBQA/kbqa/docker-compose.yml (hunk 6 above)
Comment (DnlKI) [reply KBQA/kbqa/docker-compose.yml:43]: Yes, should use the ${RESOURCE_PATH} var

**Review by smeyer198 — APPROVED** [review]

**Pull request merged**